app/assets/javascripts/modules/sticky-element-container.js

```diff
@@ -0,0 +1,136 @@
+/*
+  This module will cause a child in the target element to:
+  - hide when the top of the target element is visible;
+  - stick to the bottom of the window while the parent element is in view;
+  - stick to the bottom of the target when the user scrolls past the bottom.
+
+  Use 'data-module="sticky-element-container"' to instantiate, and add
+  `[data-sticky-element]` to the child you want to position.
+*/
+(function (Modules, root) {
+  'use strict';
+
+  var $ = root.$;
+  var $window = $(root);
+
+  Modules.StickyElementContainer = function () {
+    var self = this;
+
+    self._getWindowDimensions = function _getWindowDimensions () {
+      return {
+        height: $window.height(),
+        width: $window.width()
+      };
+    };
+
+    self._getWindowPositions = function _getWindowPositions () {
+      return {
+        scrollTop: $window.scrollTop()
+      };
+    };
+
+    self.start = function start ($el) {
+      var $element = $el.find('[data-sticky-element]');
+      var _hasResized = true;
+      var _hasScrolled = true;
+      var _interval = 50;
+      var _isDesktop = true;
+      var _windowVerticalPosition = 1;
+      var _startPosition, _stopPosition;
+
+      initialise();
+
+      function initialise () {
+        $window.resize(onResize);
+        $window.scroll(onScroll);
+        setInterval(checkResize, _interval);
```

Comment:
Good points!

`setInterval` was carried over from the previous implementation (https://github.com/alphagov/whitehall/blob/master/app/assets/javascripts/application/back-to-content.js); I know about the perf trade-offs between it and `setTimeout`.

I actually wanted to use a `requestAnimationFrame` (with a `_.throttle` on the callback) call instead of either an interval or a timeout, but since we had no polyfill in place for it I decided to just leave it as is for now.

If you want to go full operational reasoning then there are some other perf wins; such as the window event listeners that are added for every call of this module and aren't delegated.

Since 99% of the time this interval callback will be called it's going to noop due to the fact that it hits a false conditional head-on, my hunch is that it won't cause performance issues.

Thanks again for the feedback. :+1: